### PR TITLE
EVG-5691 implement new task-planner process (step 1)

### PR DESCRIPTION
**Pull request by StuartButler**

### CODE REVIEW

File: scheduler/wrapper.go [hunk 1]
```diff
@@ -56,6 +56,9 @@ func PlanDistro(ctx context.Context, conf Configuration, s *evergreen.Settings)
 	taskFindingBegins := time.Now()
 	finder := GetTaskFinder(conf.TaskFinder)
 	tasks, err := finder(distro.Id)
+	// version := distro.FinderSettings.Version
+	// finder := GetTaskFinder(version)
+	// tasks, err := finder(distro.Id)
```
Comment (tychoish) [inline scheduler/wrapper.go]: let's delete the comment.

File: globals.go [hunk 2]
```diff
@@ -136,6 +136,12 @@ const (
 	PlannerVersionLegacy  = "legacy"
 	PlannerVersionTunable = "tunable"
 
+	FinderVersionLegacy    = "legacy"
+	FinderVersionParallel  = "parallel"
+	FinderVersionPipeline  = "pipeline"
+	FinderVersionAlternate = "alternate"
+	// FinderVersionTunable = "tunable"
```
Comment (tychoish) [inline globals.go]: are we going to write this task finder?

File: globals.go [hunk 3]
```diff
@@ -346,6 +352,14 @@ var (
 		PlannerVersionTunable,
 	}
 
+	// Set of valid FinderSettings.Version strings that can be user set via the API
+	ValidFinderVersions = []string{
+		FinderVersionLegacy,
+		// FinderVersionParallel,
+		// FinderVersionPipeline,
+		// FinderVersionAlternate,
```
Comment (tychoish) [inline globals.go]: might be ok to make these options?

File: scheduler/wrapper.go (hunk 1 above)
Comment (StuartButler) [reply scheduler/wrapper.go]: Done

File: globals.go (hunk 2 above)
Comment (StuartButler) [reply globals.go]: Removing

File: globals.go (hunk 3 above)
Comment (StuartButler) [reply globals.go]: Uncommented

File: rest/model/distro.go [hunk 4]
```diff
@@ -70,7 +72,50 @@ func (s *APIPlannerSettings) ToService() (interface{}, error) {
...
+	}
+
+	return nil
+}
+
+// ToService returns a service layer distro.FinderSettings using the data from APIFinderSettings
+func (s *APIFinderSettings) ToService() (interface{}, error) {
+	settings := distro.FinderSettings{}
+	settings.Version = FromAPIString(s.Version)
+	if len(settings.Version) == 0 {
```
Comment (bsamek) [inline rest/model/distro.go]: nit: if `settings.Version == ""` (checking the length means it's less obvious that this is a string, and Go programmers have strong expectations about checking the zero values of primitive types)

Comment (StuartButler) [reply rest/model/distro.go]: Fixing

**Pull request merged**